### DISCUSSION TITLE
intersect plugin fails when using css scroll snap

### DISCUSSION THREAD

**Discussion by Ravenna** [alpinejs/alpine#2597] [7. Bugs (archived)] [answered]
I have a site where I am trying to detect as the user scrolls section to section.  When they reach a new section, we update a little scroll indicator. 

This works fine if I have scroll snap disabled. But the moment I enable the scroll-snap the x-intersect fails to pickup on the scroll.

For reference I am using:
`        scroll-snap-type: y mandatory;
        scroll-snap-points-y: repeat(100vh);`

scroll snap css.

**Comment by SimoTod** [comment]
Have you tried x-intersect:enter? It has a slightly different implementation. I noticed x-intersect alone doesn't work in some scenarios

**Reply by Ravenna** [reply]
@SimoTod I had not but I just did and with no improvement.  Is there anything else to try?

**Reply by SimoTod** [reply]
Can you create a codepen to replicate? I tried and it seemed to work.

**Comment by SimoTod** [comment, marked as answer]
I believe your issue is the browser thinking that the next div is already intersecting so the callback doesn't fire when you expect. 
Assuming each element uses the full height and given you snap scroll each one, I would use the half modifier to make sure it triggers only when it really needs to. See https://codepen.io/SimoTod/pen/jOGgyJv
Let me know if it works for your use case.

**Comment by Ravenna** [comment]
Interesting yours works.  I made one for the same reason: to test.  [Codepen](https://codepen.io/ravennainteractive/pen/qBPePMJ)

I'll review yours more closely.

**Reply by SimoTod** [reply]
Your codepen didn't have alpine and the intersect plugin. Once added, it kinda worked but I was still getting some weird results sometimes. It looks like the way to go is x-intersect:enter.half: https://codepen.io/SimoTod/pen/OJxKQVG

**Comment by Ravenna** [comment]
@SimoTod thank you for your help.  It is working how I wanted.
